yt_dlp/extractor/itprotv.py

```diff
@@ -0,0 +1,174 @@
+            'description': "Did you know that preparation is one of the most important aspects of a successful job interview? Just as the interviewer will do their research on you, it's equally as important for you to do your research on the company. In this episode Jo will share with you some key preparation tips and best practices that will hopefully help you nail your next job interview!",
+            'duration': 267,
+            'series': 'Beyond Tech',
+            'series_id': 'beyond-tech',
+            'availability': 'needs_auth',
+            'chapter': 'Job Development',
+            'chapter_number': 2,
+            'chapter_id': '5f7c78d424330c000edf04d9'
+        },
+
+    }
+    ]
+
+    def _get_course_api_json(self, webpage, course_name):
+        jwt = self._fetch_jwt(webpage)
+
+        if jwt:
+            headers = {'Authorization': 'Bearer ' + jwt}
+            course_api = self._download_json(
+                f'https://api.itpro.tv/api/urza/v3/consumer-web/course?url={course_name}&brand=00002560-0000-3fa9-0000-1d61000035f3',
+                course_name, headers=headers, note='Fetching data from course API')
+
+        return course_api
+
+    def _get_episode_api_json(self, webpage, episode_id):
+        jwt = self._fetch_jwt(webpage)
+
+        if jwt:
+            headers = {'Authorization': 'Bearer ' + jwt}
+            episode_api = self._download_json(
+                f'https://api.itpro.tv/api/urza/v3/consumer-web/brand/00002560-0000-3fa9-0000-1d61000035f3/episode?url={episode_id}',
+                episode_id, headers=headers, note='Fetching data from episode API')
+
+        return episode_api
+
+    def _fetch_jwt(self, webpage):
+        return self._search_regex(r'{"passedToken":"([A-Za-z0-9-_]*\.[A-Za-z0-9-_]*\.[A-Za-z0-9-_]+)",', webpage, 'jwt')
+
+    def _check_if_logged_in(self, webpage):
+        if '{ member: null' in webpage:
+            raise self.raise_login_required(method='cookies')
+
+    def _real_extract(self, url):
+        QUALITIES = qualities(['low', 'medium', 'high', 'veryhigh'])
+
+        episode_id = self._match_id(url)
+        webpage = self._download_webpage(url, episode_id)
+        self._check_if_logged_in(webpage)
+
+        course_name = re.match(r'https?://.*/course/(?P<course_name>[0-9a-z-]+)/.*', url).group(1)
+        course_api = self._get_course_api_json(webpage, course_name)
+        course = course_api['course']
+
+        episode_api = self._get_episode_api_json(webpage, episode_id)
+        episode = episode_api['episode']
+
+        for i in range(len(course.get('topics'))):
+            if traverse_obj(course, ('topics', i, 'id'), expected_type=str) == episode.get('topic'):
+                chapter_name = traverse_obj(course, ('topics', i, 'title'), expected_type=str)
+                chapter_id = traverse_obj(course, ('topics', i, 'id'), expected_type=str)
+                chapter_number = i + 1
```

Comment:
```suggestion
        chapter_number, chapter = next((
            i, topic for i, topic in enumerate(course.get('topics') or [], 1)
            if traverse_obj(topic, 'id') == episode.get('topic')), {})
```
then u can return 
```py
'chapter': str_or_none(chapter.get('title')),
...
```

Reply:
This is out of my limited Python knowledge it seems and I am getting a syntax error when trying to execute it. Any documentation you could point me to?

Reply:
My suggestion was missing a bracket. Should work now